### PR TITLE
[linux] add rootless build

### PR DESCRIPTION
**Pull request by lrusak**
This does a few things.
 - Creates an initial CMake build. This is now the standard in Qt6 instead of qmake. This is likely not working properly for Windows or MacOS builds at this time.
 - Adds a linux build that uses UDisks2 via dbus. This allows the application to run rootless and will prompt for an administrator password (through UDisks2 interactive authentication).
 - Adds a github actions build for the new linux build.
 - Cleans up the linux helper scripts to use the new CMake build.

The rootless build is needed to be able to run on wayland.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+if(WIN32)
+    target_sources(LibreELEC.USB-SD.Creator PUBLIC
+        deviceenumerator_windows.cpp deviceenumerator_windows.h
+        diskwriter_windows.cpp diskwriter_windows.h
+        privileges.h
+    )
+
+    target_compile_options(LibreELEC.USB-SD.Creator
+        -Ic:\Qt\Qt5.6.1\Tools\mingw492_32\i686-w64-mingw32\include
```
Comment (kambala-decapitator) [inline CMakeLists.txt:153]: nice :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTOUIC ON)
+
+find_package(QT NAMES Qt6 REQUIRED COMPONENTS Core)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Gui Network Widgets LinguistTools)
+
+qt_add_executable(LibreELEC.USB-SD.Creator WIN32 MACOSX_BUNDLE
```
Comment (kambala-decapitator) [inline CMakeLists.txt:12]: I'd use a shorter target name

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        "/"
+    FILES
+        ${resources2_resource_files}
+)
+
+if(static)
+    target_compile_definitions(LibreELEC.USB-SD.Creator PUBLIC
+        STATIC
+    )
+endif()
```
Comment (kambala-decapitator) [inline CMakeLists.txt:143]: what is this for?

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+if(MACOS)
+    target_sources(LibreELEC.USB-SD.Creator PUBLIC
+        privileges_unix.cpp privileges_unix.h
+        diskwriter_unix.cpp diskwriter_unix.h
+        deviceenumerator_unix.cpp deviceenumerator_unix.h
+    )
+endif()
+
+if(LINUX)
```
Comment (kambala-decapitator) [inline CMakeLists.txt:165]: New in version 3.25.

File: creator.cpp [hunk 5]
```diff
@@ -222,9 +229,7 @@ Creator::Creator(Privileges &privilegesArg, QWidget *parent) :
 
 bool Creator::showRootMessageBox()
 {
-#ifdef Q_OS_WIN
-    return false;   // always run with administrative privileges
-#else
+#ifdef Q_OS_MAC
```
Comment (kambala-decapitator) [inline creator.cpp:232]: what about BSD?

File: creator.h [hunk 6]
```diff
@@ -33,7 +33,7 @@
 #include "movingaverage.h"
 #include "translator.h"
 
-#ifdef Q_OS_UNIX
+#ifdef Q_OS_MAC
```
Comment (kambala-decapitator) [inline creator.h:36]: what about BSD?

File: diskwriter_udisks2.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+#pragma once
+
+#include "diskwriter.h"
+#include <QFile>
+#include <QtDBus>
+
+class DiskWriter_udisks2 : public DiskWriter
+{
+    Q_OBJECT
```
Comment (kambala-decapitator) [inline diskwriter_udisks2.h:27]: the macro shouldn't be needed as you don't have any signals

File: main.cpp [hunk 8]
```diff
@@ -20,7 +20,7 @@
 #include "creator.h"
 #include "version.h"
 
-#ifdef Q_OS_UNIX
+#ifdef Q_OS_MAC
```
Comment (kambala-decapitator) [inline main.cpp:23]: what about BSD?

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

**Comment by chewitt** [comment]
@kambala-decapitator there are no plans to support the app on BSD

**Comment by kambala-decapitator** [comment]
funny, I can't resolve my own conversations :)

> @kambala-decapitator there are no plans to support the app on BSD

clear, thanks!

File: CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    set_target_properties(LibreELEC.USB-SD.Creator PROPERTIES SUFFIX .Linux.bin)
+
+    find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS DBus)
```
Comment (fuzzard) [inline CMakeLists.txt:168]: Ill mention, but i dont have references off hand, however multiple find_package calls to the same thing (ie Qt${QT_VERSION_MAJOR}) wont actually work.

The way the caching of find_package works, it essentially wont change anything after its called once.

Youd be better making the linux check early, and adding a variable for this to the initial call.

**Comment by lrusak** [comment]
Just an FYI for those interested. In the OP it does say:
> This is likely not working properly for Windows or MacOS builds at this time.

**Comment by kambala-decapitator** [comment]
we can merge as is and provide adjustments in the following PRs

**Pull request merged**